libarchive/archive_write_set_format_pax.c

```diff
@@ -329,17 +329,42 @@ add_pax_attr_binary(struct archive_string *as, const char *key,
 	archive_strappend_char(as, '\n');
 }
 
+static void
+archive_write_pax_header_xattr(struct pax *pax, const char *encoded_name,
+    const void *value, size_t value_len)
+{
+	struct archive_string s;
+	char *encoded_value;
+
+	if (1) {
```

Comment:
There are two possible routes for writing SCHILY.xattr:
  * Add an option to the writer.  This is routine and something we could do today.
  * Write both.  Apart from size concerns, this is a pretty appealing idea.

`archive_write_set_format_gnutar` doesn't actually support xattrs today.  Adding it would be a bit of work since that code only handles the old gnu tar format (which does not support xattrs at all), not the newer pax-based format.  I suspect the best route forward there is to add an `archive_write_set_format_gnutar_pax` that just calls `archive_write_set_format_pax` and then sets a couple of options to tweak the output details.